src/mds/MDCache.cc

```diff
@@ -1418,6 +1423,49 @@ void MDCache::adjust_subtree_after_rename(CInode *diri, CDir *olddir, bool pop)
   show_subtrees();
 }
 
+// ===================================
+// Default cache memory limit
+
+
+/*
+ * Get the default cache memory limit using the mds_memory_target
+ */
+uint64_t MDCache::get_default_mds_cache_memory_limit()
+{
+  // RSS usage has been experimentally shown to be around 25 percent more than the cache_memory_limit. Hence, the cache_memory_limit is reset as per the mds_memory_target(which is an upper bound on the rss usage)
+  uint64_t temp_cache_limit = (uint64_t) (mds->memory_target/1.25);
+  mds->cct->_conf.set_val_default("mds_cache_memory_limit", stringify(temp_cache_limit));
+  return temp_cache_limit;
+}
+
+// ===================================
+// Cache Autotuning
+
+
+/*
+ * If mds_cache_autotune is set to true, then adjust cache_memory_limit using the rss usage periodically as per the mds_cache_resize_interval
+ */
+void MDCache::resize()
+{
+    if (cache_autotune) {
+      auto now = clock::now();
+      auto cache_resize_interval = g_conf().get_val<std::chrono::seconds>("mds_cache_resize_interval"); 
+      if (cache_memory_limit > mds->memory_target)
+	// RSS usage has been experimentally shown to be around 25 percent more than the cache_memory_limit. Hence when the mds_memory_target is changed and the cache_memory_limit becomes more than the memory_target, the cache_memory_limit is reset as per the mds_memory_target(which is an upper bound on the rss usage) 
+	cache_memory_limit = mds->memory_target/1.25;
+      if (cache_resize_interval == std::chrono::duration<long int>(0) || ((now-last_cache_resize_time) > cache_resize_interval)) {
+	MemoryModel::snap last;
+        mm.sample(&last);
+        if (((uint64_t)(last.get_rss())) > mds->memory_target) 
+          cache_memory_limit *= 0.99;
```

Comment:
This is doing floating point arithmetic on a `uint64_t`. Let's use integer arith instead:

`cache_memory_limit = cache_memory_limit/100*99.`